src/extension.ts

```diff
@@ -109,4 +114,94 @@ export const activate = () => {
         if (!getExtensionSetting('runOnSave') || reason === vscode.TextDocumentSaveReason.AfterDelay) return
         waitUntil(performFixes())
     })
+
+    vscode.workspace.onDidChangeTextDocument(
+        ({ contentChanges, document, reason }) => {
+            if (!getExtensionSetting("insertMissingCommaOnEnter")) {
+                return;
+            }
+
+            if (
+                !vscode.languages.match(['json', 'jsonc'], document)
+            ) {
+                return;
+            }
+
+            if (contentChanges.length === 0) {
+                return;
+            }
+
+            contentChanges = [...contentChanges].sort((a, b) => a.range.start.compareTo(b.range.start))
+
+            const editor = vscode.window.activeTextEditor;
+
+            if (
+                document.uri !== editor?.document.uri ||
+                ["output"].includes(editor.document.uri.scheme)
+            ) {
+                return;
+            }
+
+            if (
+                vscode.workspace.fs.isWritableFileSystem(
+                    document.uri.scheme
+                ) === false
+            ) {
+                return;
+            }
+
+            if (oneOf(reason, vscode.TextDocumentChangeReason.Undo, vscode.TextDocumentChangeReason.Redo)) {
+                return
+            }
+
+            if (contentChanges.some((change) => !isContainEoL(change.text))) {
+                return;
+            }
+
+            void editor.edit((edit) => {
+                for (const [i, change] of contentChanges.entries()) {
+                    const prevLinePosition = change.range.start;
+
+                    const prevLine = document.lineAt(prevLinePosition.line + i);
+                    const prevLineText = prevLine.text;
+
+                    const prevLineLastChar = prevLineText.at(-1);
+
+                    if (!prevLineLastChar) {
+                        continue;
+                    }
+
+                    if (startsWithComment(prevLine.text.trim())) {
+                        continue;
+                    }
+
+                    const textWithouComments = stripJsonComments(document.getText());
```

Comment:
```suggestion
                    const textWithoutComments = stripJsonComments(document.getText());
```

Reply:
Also let's move this out of the loop as we process absolutely the same text for each cursor position

Reply:
I made some improvements so that the comment stripping function is called only once and only after all conditions to minimize its call